### PR TITLE
Adds capability to store `pkt_list` for each session packet

### PR DESCRIPTION
**Pull request by piersoh**
This PR adds a capability to store `pkt_list` for each session packet which controlled via the `store` parameter to `sniff()`. When Scapy is run in session mode it parses the provided pcap/packetlist into "session" packets which have len=None. This PR allows for the storage of the original packets that make up the "session" packets in an attached `PacketList` as a class variable `pkt_list`. It is something I have found useful when parsing TCP/HTTP flows.

It was done fairly quickly so happy to be advised on how best to update/modify for better compliance. e.g. I wasn't sure if one should add a field to the `Packet` type for this or somewhere else?

### CODE REVIEW

**Comment by gpotter2** [comment]
Could you explain what's the use case for this, and add a test that uses it?
I also don't see how some parts work.

File: scapy/sessions.py [hunk 1]
```diff
@@ -315,6 +317,8 @@ def _process_packet(self, pkt):
         # Add the data to the buffer
         # Note that this take care of retransmission packets.
         data.append(new_data, seq)
+        if self.store:
```
Comment (gpotter2) [inline scapy/sessions.py:320]: Where is  `store` defined?

File: scapy/sessions.py [hunk 2]
```diff
@@ -332,13 +336,16 @@ def _process_packet(self, pkt):
...
-            metadata.clear()
             del self.tcp_frags[ident]
             pay.underlayer.remove_payload()
             if IP in pkt:
                 pkt[IP].len = None
                 pkt[IP].chksum = None
-            return pkt / packet
+            pkt_ret = pkt / packet
+            if self.store:
+                pkt_ret.pkt_list=metadata["pkt_list"]
```
Comment (gpotter2) [inline scapy/sessions.py:346]: You gotta document this somewhere if you want this to be added. Adding a mysterious packet field isn't great

Comment (piersoh) [reply scapy/sessions.py:346]: The use case is when analysing [HTTP] packet flows using sniff(...`session=TCPSession`) which results in each request and response being extracted into 'session' packets (len=None). The proposed `pkt_list` maintains a `PacketList` of the actual packets that makes up each 'session' packet. This list of actual packets are useful for analysing the data flow rates at packet level granularity per response - this is useful when the responses can be quite large like media flows which can be constructed from thousands of actual packets. This is particularly useful now modern browsers pool TCP connections so different [media] type request/responses can be spread across different TCP sessions and the `pkt_list` allows for per media packet level analysis.

I wasn't sure what was the best approach as to choosing how to 'attach' the `pkt_list` to the session packets. I'd be happy to document it if people are happy with the approach and I'm open to suggestions as to better place to store the `pkt_list`. Also happy to fix any PEP8 issues.

**Comment by piersoh** [comment]
I've updated the code to be PEP8 compliant. The `self.store` is just using the existing `store` variable/property to activate the `pkt_list` functionality. As I mentioned I can document this feature if the implementation is agreeable - if not maybe you could suggest a better place/way to store the packet list.

**Comment by gpotter2** [comment]
Sessions have been changed a lot so this PR is obsolete. Closing.

**Pull request closed without merging**